### ISSUE TITLE
psake doesn't appear to use the Roslyn compiler when running msbuild

### ISSUE THREAD

**Issue by TrevorPilley** [psake/psake#195]
In Visual Studio 2015, it is possible to use some C# 6 features in projects targeting older versions of .NET (e.g.):

```
public IList<string> Names { get; } = new List<string>();
```

```
throw new ArgumentNullException(nameof(param));
```

When using these languate features in a .NET 4.0 project, Visual Studio 2015 builds successfully, however using psake results in compilation errors. This leads me to suspect that psake isn't using the Roslyn compiler - is that correct? and if so, could it be updated to use Roslyn if installed over the older msbuild?

**Comment by knunery** [comment]
What version of psake are you using? This can be fixed in most cases by adding`Framework "4.6"` at the beginning of your default.ps1 file.

**Comment by jozefizso** [comment]
We use psake in our builds and with `Framework "4.6"` we have no issues build with latest Visual Studio toolset with support for C# 6.
